src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -0,0 +1,1425 @@
+        >>> outputs = model(**inputs)
+
+        >>> for output in outputs:  # loop over sets of text queries
+        >>>     last_hidden_state = output.last_hidden_state
+        >>>     pooled_output = output.pooler_output  # pooled (EOS token) states
+        ```"""
+        batch_size = input_ids.shape[0]
+
+        # Get embeddings for all text queries in all batch samples
+        if attention_mask is not None:
+            output = tuple(
+                [
+                    self.text_model(
+                        input_ids=input_ids[idx],
+                        attention_mask=attention_mask[idx],
+                        output_attentions=output_attentions,
+                        output_hidden_states=output_hidden_states,
+                        return_dict=return_dict,
+                    )
+                    for idx in range(batch_size)
+                ]
+            )
+        else:
+            output = tuple(
+                [
+                    self.text_model(
+                        input_ids=input_ids[idx],
+                        attention_mask=None,
+                        output_attentions=output_attentions,
+                        output_hidden_states=output_hidden_states,
+                        return_dict=return_dict,
+                    )
+                    for idx in range(batch_size)
+                ]
+            )
+        return output
+
+
+class OwlViTVisionTransformer(nn.Module):
+    def __init__(self, config: OwlViTVisionConfig):
+        super().__init__()
+        self.config = config
+        embed_dim = config.hidden_size
+
+        self.embeddings = OwlViTVisionEmbeddings(config)
+        self.pre_layrnorm = nn.LayerNorm(embed_dim)
+        self.encoder = OwlViTEncoder(config)
+        self.post_layernorm = nn.LayerNorm(embed_dim)
+
+    @add_start_docstrings_to_model_forward(OWLVIT_VISION_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=BaseModelOutputWithPooling, config_class=OwlViTVisionConfig)
+    def forward(
+        self,
+        pixel_values: Optional[torch.FloatTensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        train: Optional[bool] = False,
+    ) -> Union[Tuple, BaseModelOutputWithPooling]:
+        r"""
+        Returns:
+
+        """
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if pixel_values is None:
+            raise ValueError("You have to specify pixel_values")
+
+        pixel_values = pixel_values.to(torch.float32)
```

Comment:
That's interesting, we don't have this in any other vision model, we assume that users input a `FloatTensor` (as explained in the docstring)

Maybe we can remove this for consistency, or add it to any other vision model?